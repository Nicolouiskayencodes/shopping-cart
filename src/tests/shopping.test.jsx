import { vi, describe, it, expect } from 'vitest'
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Shopping from '../components/shopping';

describe("Shopping component", () => {
  
  it("should call the prop function when clicked", async () => {
    const onClick = vi.fn();
    const user = userEvent.setup()
    render(<Shopping props={[{id:1}]} add={onClick} />);
    await user.click(screen.getAllByRole("button")[0])
    const button = screen.getByRole("button", { name: "Add to Cart" });

    await user.click(button);

    expect(onClick).toHaveBeenCalled();
  });

  it("should not call the prop function when it isn't clicked", async () => {
    const onClick = vi.fn();
    const user = userEvent.setup()
    render(<Shopping props={[{id: 1}]} add={onClick} />);
    await user.click(screen.getAllByRole("button")[0])
    expect(onClick).not.toHaveBeenCalled();
  });
  it("renders card", ()=>{
    const { container } = render(<Shopping props={[{id: 1}]} add="function"/>);
    expect(container).toMatchSnapshot();
  })
});
