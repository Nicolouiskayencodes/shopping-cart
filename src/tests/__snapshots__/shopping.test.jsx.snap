// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shopping component > renders card 1`] = `
<div>
  <div>
    <div
      class="_container_56a2b4"
    >
      <button
        class="_btn_56a2b4"
      >
         
        <div
          class="_card_71d7ae"
        >
          <img
            class="_img_71d7ae"
          />
          <h2
            class="_name_71d7ae"
          />
          <p
            class="_price_71d7ae"
          >
            $
          </p>
        </div>
      </button>
    </div>
  </div>
</div>
`;
